test/MarketContractOraclize.js

```diff
@@ -179,16 +179,407 @@ contract('MarketContractOraclize', function(accounts) {
+        unsignedOrderValues = [0, 0, entryOrderPrice, timeStamp, 1];
+        orderQty = 5;   // user is attempting to buy 5
+        orderHash = await orderLib.createOrderHash.call(
+            MarketContractOraclize.address,
+            orderAddresses,
+            unsignedOrderValues,
+            orderQty
+        );
+
+        // transfer half of the collateral tokens to the second account.
+        initBalance = await collateralToken.INITIAL_SUPPLY.call().valueOf();
+        var balanceToTransfer = initBalance / 2;
+        await collateralToken.transfer(accounts[1], balanceToTransfer, {from: accounts[0]});
+
+        // create approval and deposit collateral tokens for trading.
+        var amountToDeposit = 5000000;
+        await collateralToken.approve(collateralPool.address, amountToDeposit, {from: accounts[0]})
+        await collateralToken.approve(collateralPool.address, amountToDeposit, {from: accounts[1]})
+
+        // move tokens to the collateralPool
+        await collateralPool.depositTokensForTrading(amountToDeposit, {from: accounts[0]})
+        await collateralPool.depositTokensForTrading(amountToDeposit, {from: accounts[1]})
+
+        makerAccountBalanceBeforeTrade = await collateralPool.getUserAccountBalance.call(accounts[0]);
+        takerAccountBalanceBeforeTrade = await collateralPool.getUserAccountBalance.call(accounts[1]);
+
+        // Execute trade between maker and taker for excess of order.
+        var qtyToFill = 7;
+        var orderSignature = utility.signMessage(web3, accountMaker, orderHash)
+
+        // trade order requesting more than available qty
+        await marketContract.tradeOrder(
+            orderAddresses,
+            unsignedOrderValues,
+            orderQty,           // qty is five
+            qtyToFill,          // let us fill 7 (excess lots)
+            orderSignature[0],  // v
+            orderSignature[1],  // r
+            orderSignature[2],  // s
+            {from: accountTaker}
+        );
+        // trade order shouldn't fail, should NetPos should be +/- 5 (qty)
+        var makerNetPos = await collateralPool.getUserPosition.call(accountMaker);
+        var takerNetPos = await collateralPool.getUserPosition.call(accountTaker);
+        assert.equal(makerNetPos.toNumber(), 5, "Maker should be long 5");
+        assert.equal(takerNetPos.toNumber(), -5, "Taker should be short 5");
+    });
+});
+
+
+contract('ContractOverCancel', async function(accounts) {
+
+    var collateralPool;
+    var marketToken;
+    var marketContract;
+    var collateralToken;
+    var orderLib;
+    var makerAccountBalanceBeforeTrade;
+    var takerAccountBalanceBeforeTrade;
+    var entryOrderPrice = 33025;
+
+    //  cancelling more than what is available to be cancelled from the order
+
+    it("Attempt to cancel more than remaining in contract, cancels the balance", async function() {
+        var collateralPool = await MarketCollateralPool.deployed();
```

Comment:
@craig-iam-smith - aren't these variables already declared above?  I think there is a variable scope issue possibly here.